extention/TodayViewController.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  TodayViewController.swift
+//  extention
+//
+//  Created by Jawaher Alagel on 7/13/20.
+//  Copyright © 2020 Jawaher Alaggl. All rights reserved.
+//
+
+import UIKit
+import NotificationCenter
+
+class TodayViewController: UIViewController, NCWidgetProviding {
+        
+    @IBOutlet weak var label: UILabel!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        if let name = UserDefaults.init(suiteName: "group.com.jawaher.widget")?.value(forKey: "name")
+        {
+            label.text = name as? String
+        }
+        if let name2 = UserDefaults.init(suiteName: "group.com.jawaher.widget")?.value(forKey: "name2")
+        {
+           label.text = name2 as? String
+        }
+    }
+        
+    func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
+        // Perform any setup necessary in order to update the view.
+        
+        // If an error is encountered, use NCUpdateResult.Failed
+        // If there's no update required, use NCUpdateResult.NoData
+        // If there's an update, use NCUpdateResult.NewData
+        
+        completionHandler(NCUpdateResult.newData)
+    }
```

Comment:
https://developer.apple.com/documentation/notificationcenter/ncwidgetproviding/1490262-widgetperformupdate

according to apple doc this function seems depreciated, try to find some other way for doing the same